client-server-protocol/src/main/scala/zio/raft/protocol/ServerMessages.scala

```diff
@@ -0,0 +1,190 @@
+ */
+case class KeepAliveResponse(
+  timestamp: Instant
+) extends ServerMessage
+
+/**
+ * Client request execution result.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param requestId Echoed request ID from ClientRequest for correlation
+ * @param result Binary result data from command/query execution
+ */
+case class ClientResponse(
+  requestId: RequestId,
+  result: ByteVector
+) extends ServerMessage
+
+/**
+ * Server-initiated work dispatch to client (one-way pattern).
+ * Server targets via ZeroMQ routing ID, no session ID needed.
+ *
+ * @param requestId Unique identifier for acknowledgment correlation
+ * @param payload Binary work payload for client processing
+ * @param createdAt Timestamp when request was created
+ */
+case class ServerRequest(
+  requestId: RequestId,
+  payload: ByteVector,
+  createdAt: Instant
+) extends ServerMessage
+
+/**
+ * Client request processing error.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param reason Error classification and details
+ * @param leaderId Optional leader ID for redirection
+ */
+case class RequestError(
+  reason: RequestErrorReason,
+  leaderId: Option[MemberId]
+) extends ServerMessage
+
+// ============================================================================
+// REASON ENUMS
+// ============================================================================
+
+/**
+ * Reasons for session rejection.
+ */
+sealed trait RejectionReason
+
+object RejectionReason {
+  /**
+   * Server is not the current Raft leader.
+   */
+  case object NotLeader extends RejectionReason
+
+  /**
+   * Requested session ID does not exist (expired or never created).
+   */
+  case object SessionNotFound extends RejectionReason
+
+  /**
+   * Client capabilities are invalid or unsupported.
+   */
+  case object InvalidCapabilities extends RejectionReason
+}
+
+/**
+ * Reasons for server-initiated session closure.
+ */
+sealed trait SessionCloseReason
+
+object SessionCloseReason {
+  /**
+   * Server is shutting down gracefully.
+   */
+  case object Shutdown extends SessionCloseReason
+
+  /**
+   * Server lost leadership and cannot serve clients.
+   */
+  case object NotLeaderAnymore extends SessionCloseReason
+
+  /**
+   * Client sent unexpected/invalid message for current session state.
+   */
+  case object SessionError extends SessionCloseReason
+
+  /**
+   * Client connection was closed by the server or OS (e.g. timeout, network error).
+   */
+  case object ConnectionClosed extends SessionCloseReason
+
+  /**
+   * Session expired due to client inactivity (no keep-alive messages).
+   */
+  case object SessionTimeout extends SessionCloseReason
+}
+
+/**
+ * Reasons for client request processing errors.
+ */
+sealed trait RequestErrorReason
+
+object RequestErrorReason {
+  /**
+   * Server is not the current Raft leader (extends NotLeader for requests).
+   */
+  case object NotLeaderRequest extends RequestErrorReason
```

Comment:
Let's remove this, we would use the SessionError instead to reject those type of requests.